core/src/com/unciv/logic/battle/ICombatant.kt

```diff
@@ -30,4 +30,6 @@ interface ICombatant {
         return this is CityCombatant
     }
     fun isCivilian() = this is MapUnitCombatant && this.unit.isCivilian()
+
+    fun getXP() = (this as MapUnitCombatant).unit.promotions.totalXpProduced()
```

Comment:
Surely this should be a function of MapUnitCombatant and not ICombatant

Reply:
That's fair since ICombatant doesn't always have XP